dspace-api/src/main/java/org/dspace/app/mediafilter/TikaTextExtractionFilter.java

```diff
@@ -0,0 +1,177 @@
+            oe.printStackTrace();
+            log.error("OutOfMemoryError occurred when extracting text from bitstream in Item {}. " +
+                          "You may wish to enable 'textextractor.use-temp-file'.", currentItem.getID().toString(), oe);
+            throw oe;
+        }
+
+        if (StringUtils.isNotEmpty(extractedText)) {
+            // if verbose flag is set, print out extracted text to STDOUT
+            if (verbose) {
+                System.out.println("(Verbose mode) Extracted text:");
+                System.out.println(extractedText);
+            }
+
+            // return the extracted text as a UTF-8 stream.
+            return new ByteArrayInputStream(extractedText.getBytes(StandardCharsets.UTF_8));
+        }
+        return null;
+    }
+
+    /**
+     * Extracts the text out of a given source InputStream, using a temporary file. This decreases the amount of memory
+     * necessary for text extraction, but can be slower as it requires writing extracted text to a temporary file.
+     * @param source source InputStream
+     * @param verbose verbose mode enabled/disabled
+     * @return InputStream for temporary file containing extracted text
+     * @throws IOException
+     * @throws SAXException
+     * @throws TikaException
+     */
+    private InputStream extractUsingTempFile(InputStream source, boolean verbose)
+        throws IOException, TikaException, SAXException {
+        File tempExtractedTextFile = File.createTempFile("dspacetextextract" + source.hashCode(), ".txt");
+
+        if (verbose) {
+            System.out.println("(Verbose mode) Extracted text was written to temporary file at " +
+                                   tempExtractedTextFile.getAbsolutePath());
+        } else {
+            tempExtractedTextFile.deleteOnExit();
+        }
+
+        // Open temp file for writing
+        try (FileWriter writer = new FileWriter(tempExtractedTextFile, StandardCharsets.UTF_8)) {
+            // Initialize a custom ContentHandlerDecorator which is a BodyContentHandler.
+            // This mimics the behavior of Tika().parseToString(), which only extracts text from the body of the file.
+            // This custom Handler writes any extracted text to the temp file.
+            ContentHandlerDecorator handler = new BodyContentHandler(new ContentHandlerDecorator() {
+                /**
+                 * Write all extracted characters directly to the temp file.
+                 */
+                @Override
+                public void characters(char[] ch, int start, int length) {
+                    try {
+                        writer.append(new String(ch), start, length);
+                    } catch (IOException e) {
+                        log.error("Could not append to temporary file at {} when performing text extraction",
+                                  tempExtractedTextFile.getAbsolutePath(), e);
+                    }
```

Comment:
Re-throw the IOException?  Otherwise the extraction continues, and the log could be flooded.